### PR TITLE
Whitelist asset registeraddress

### PR DESCRIPTION
**Pull request by lawlawlaw**
New user's address registration requires whitelist asset. Zero fee for the transaction.

### CODE REVIEW

File: src/policy/policylist.cpp [hunk 1]
```diff
@@ -41,9 +41,10 @@ CPolicyList::base::size_type CPolicyList::size(){
 }
 
 //Add to the sorted list
-void CPolicyList::add_sorted(CKeyID* id){
+bool CPolicyList::add_sorted(CKeyID* id){
   boost::recursive_mutex::scoped_lock scoped_lock(_mtx);
   base::insert(*id);
+  return find(id);
```
Comment (nkostoulas) [inline src/policy/policylist.cpp]: think insert already returns an iterator and a bool for success, no need for the additional find

**Comment by lawlawlaw** [comment]
FYI I'm removing the exception throws from whitelist.cpp.

File: src/policy/policylist.cpp (hunk 1 above)
Comment (lawlawlaw) [reply src/policy/policylist.cpp]: Check no longer needed - removed.

**Comment by lawlawlaw** [comment]
I've kept some throws but simplified them into one basic category (std::invalid_argument), and handled where necessary.

**Pull request merged**